Kernel/Bus/PCI/Controller/GenericECAMHostController.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2023-2025, Sönke Holz <sholz8530@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Boot/CommandLine.h>
+#include <Kernel/Bus/PCI/Access.h>
+#include <Kernel/Bus/PCI/Controller/MemoryBackedHostBridge.h>
+#include <Kernel/Bus/PCI/DeviceTreeHelpers.h>
+#include <Kernel/Firmware/DeviceTree/DeviceTree.h>
+#include <Kernel/Firmware/DeviceTree/Driver.h>
+#include <Kernel/Firmware/DeviceTree/Management.h>
+
+namespace Kernel::PCI {
+
+class GenericDeviceTreeECAMHostController final : public MemoryBackedHostBridge {
+public:
+    static ErrorOr<NonnullOwnPtr<GenericDeviceTreeECAMHostController>> create(DeviceTree::Device const& device)
+    {
+        auto domain = TRY(determine_pci_domain_for_devicetree_node(device.node(), device.node_name()));
+        auto configuration_space = TRY(device.get_resource(0));
+
+        if (configuration_space.size < memory_range_per_bus * (domain.end_bus() - domain.start_bus() + 1))
+            return ERANGE;
+
+        return adopt_nonnull_own_or_enomem(new (nothrow) GenericDeviceTreeECAMHostController(domain, configuration_space.paddr));
+    }
+
+protected:
+    GenericDeviceTreeECAMHostController(Domain const& domain, PhysicalAddress physical_address)
+        : MemoryBackedHostBridge(domain, physical_address)
+    {
+    }
+};
+
+static constinit Array const compatibles_array = {
+    "pci-host-ecam-generic"sv,
```

Comment:
I could have sworn we had that one riscv one listed as well, but that might have been one of your hack branches